ori/bucket/server/suite_test.go

```diff
@@ -0,0 +1,271 @@
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	//+kubebuilder:scaffold:imports
+)
+
+const (
+	slowSpecThreshold    = 20 * time.Second
+	eventuallyTimeout    = 20 * time.Second
+	pollingInterval      = 250 * time.Millisecond
+	consistentlyDuration = 1 * time.Second
+	apiServiceTimeout    = 5 * time.Minute
+
+	bucketBaseURL = "example.com"
+)
+
+var (
+	srvCtx context.Context
+	cancel context.CancelFunc
+
+	bucketClient v1alpha1.BucketRuntimeClient
+
+	testEnv    *envtest.Environment
+	testEnvExt *envtestutils.EnvironmentExtensions
+	cfg        *rest.Config
+	k8sClient  client.Client
+	rookConfig *rook.Config
+)
+
+func TestAPIs(t *testing.T) {
+	SetDefaultConsistentlyPollingInterval(pollingInterval)
+	SetDefaultEventuallyPollingInterval(pollingInterval)
+	SetDefaultEventuallyTimeout(eventuallyTimeout)
+	SetDefaultConsistentlyDuration(consistentlyDuration)
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Bucket GRPC Server Suite")
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true), zap.Level(zapcore.InfoLevel)))
+
+	var err error
+
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			filepath.Join(modutils.Dir("github.com/rook/rook", "deploy", "examples", "crds.yaml")),
+		},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	testEnvExt = &envtestutils.EnvironmentExtensions{
+		APIServiceDirectoryPaths: []string{
+			modutils.Dir("github.com/onmetal/onmetal-api", "config", "apiserver", "apiservice", "bases"),
+		},
+		ErrorIfAPIServicePathIsMissing: true,
+	}
+
+	cfg, err = envtestutils.StartWithExtensions(testEnv, testEnvExt)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	DeferCleanup(envtestutils.StopWithExtensions, testEnv, testEnvExt)
+
+	Expect(rookv1.AddToScheme(scheme.Scheme)).To(Succeed())
+	Expect(storagev1alpha1.AddToScheme(scheme.Scheme)).To(Succeed())
+	Expect(bucketv1alpha1.AddToScheme(scheme.Scheme)).To(Succeed())
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+	SetClient(k8sClient)
+
+	apiSrv, err := apiserver.New(cfg, apiserver.Options{
+		MainPath:     "github.com/onmetal/onmetal-api/cmd/onmetal-apiserver",
+		BuildOptions: []buildutils.BuildOption{buildutils.ModModeMod},
+		ETCDServers:  []string{testEnv.ControlPlane.Etcd.URL.String()},
+		Host:         testEnvExt.APIServiceInstallOptions.LocalServingHost,
+		Port:         testEnvExt.APIServiceInstallOptions.LocalServingPort,
+		CertDir:      testEnvExt.APIServiceInstallOptions.LocalServingCertDir,
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("starting the onmetal-api aggregated api server")
+	Expect(apiSrv.Start()).To(Succeed())
+	DeferCleanup(apiSrv.Stop)
+
+	Expect(envtestutils.WaitUntilAPIServicesReadyWithTimeout(apiServiceTimeout, testEnvExt, k8sClient, scheme.Scheme)).To(Succeed())
```

Comment:
As stated above: as we are only working against the Rook API here we don't need the `onmetal-api` server in this test suite.

Reply:
Since  [here](https://github.com/onmetal/cephlet/blob/6981634ddcdc0f4bee1f36dcbe2da03f8c4ac95a/ori/bucket/server/suite_test.go#L212) we need to create a `BucketClass`  which is of type `github.com/onmetal/onmetal-api/api/storage/v1alpha1`, hence we used onmetal api server.